DOC: Add links and comments to script repository (#5737)
Slicer/Slicer: Docs/developer_guide/script_repository.md

:::

:::{note}

> **Fully worked examples**:

Reviewer: What do you mean by "fully worked"? Complete examples?

Author: Well I meant the opposite of "snippet", so it could be changed to "Use in context" maybe just "More code for reference"

Reviewer: "More code for reference" sounds good to me.

Author: Updated the name

> * The [Slicer source code](https://github.com/Slicer/Slicer) is about half written in Python and can be used to find working examples

Reviewer 2: ```suggestion
The Slicer source code has Python [scripted modules](https://github.com/Slicer/Slicer/tree/master/Modules/Scripted) and [scripted Segmentation Editor effects](https://github.com/Slicer/Slicer/tree/master/Modules/Loadable/Segmentations/EditorEffects/Python) that can be used as working examples.
```

It turns that ~10% of Slicer is written in python:
![image](https://user-images.githubusercontent.com/219043/125210291-e3bb9980-e26c-11eb-89ac-0fcf91d298c2.png)

```
Modules/Scripted/DICOMPlugins
Modules/Scripted/DICOMLib
Modules/Scripted/SampleData
Modules/Scripted/DICOM
Modules/Scripted/SegmentEditor
Modules/Scripted/SegmentStatistics
Modules/Scripted/Endoscopy
Modules/Scripted/DICOMPatcher
Modules/Scripted/ScreenCapture
Modules/Scripted/CropVolumeSequence
Modules/Scripted/DataProbe
Modules/Scripted/LabelStatistics
Modules/Scripted/VectorToScalarVolume
```

> * Most [Slicer Extensions](https://github.com/topics/3d-slicer-extension) are written in Python to address specific use cases.  Looking at their source code can be valuable.

Reviewer 2: ```suggestion
* Most [Slicer Extensions](https://github.com/topics/3d-slicer-extension) are written in Python to address specific use cases.  Looking at their source code can be informative.
```

> * [The Slicer Discourse forum](https://discourse.slicer.org/) has many code snippets and discussion. 

Reviewer 2: ```suggestion
* The [Slicer Discourse forum](https://discourse.slicer.org/) has many code snippets and discussions. 
```
